### PR TITLE
Make use of nice new tech for the new spatialite layer dialog

### PR DESCRIPTION
**Pull request by nirvn**
## Description

Make use of nice new tech for the new spatialite layer dialog:
- Use provider connection combo box to list connections
- Use database provider connection functions to store a new connection

In action:
Peek 2020-03-16 11-44

### CODE REVIEW

File: src/app/qgsnewspatialitelayerdialog.cpp [hunk 1]
```diff
@@ -274,10 +244,22 @@ void QgsNewSpatialiteLayerDialog::selectionChanged()
...
 {
-  QString dbPath = mDatabaseComboBox->currentText();
+  QString dbPath = QFileDialog::getSaveFileName( this, tr( "New SpatiaLite Database File" ),
+                   QDir::homePath(),
+                   tr( "SpatiaLite" ) + " (*.sqlite *.db *.sqlite3 *.db3 *.s3db)", nullptr, QFileDialog::DontConfirmOverwrite );
+
   if ( dbPath.isEmpty() )
     return false;
 
+  if ( !dbPath.endsWith( QLatin1String( ".sqlite" ), Qt::CaseInsensitive ) &&
```
Comment (nyalldawson) [inline src/app/qgsnewspatialitelayerdialog.cpp]: Could use QgsFileUtils::ensureFileNameHasExtension here (that supports a list of valid extensions)

File: src/app/qgsnewspatialitelayerdialog.cpp [hunk 2]
```diff
@@ -75,20 +78,7 @@ QgsNewSpatialiteLayerDialog::QgsNewSpatialiteLayerDialog( QWidget *parent, Qt::W
...
-  mDatabaseComboBox->clear();
-  while ( it != keys.end() )
-  {
-    // retrieving the SQLite DB name and full path
-    QString text = settings.value( *it + "/sqlitepath", "###unknown###" ).toString();
-    mDatabaseComboBox->addItem( text );
-    ++it;
-  }
-  settings.endGroup();
+  mDatabaseComboBox->setModel( new QgsProviderConnectionModel( QStringLiteral( "spatialite" ), this ) );
```
Comment (nyalldawson) [inline src/app/qgsnewspatialitelayerdialog.cpp]: You could just use QgsProviderConnectionComboBox here directly -- that gives you the setConnection method you are missing in createDb

**Review by nyalldawson — COMMENTED** [review]
Nice cleanup!

**Comment by uclaros** [comment]
Am I the only one that thinks the new spatialite and new geopackage dialogs should offer the same UX?

**Comment by nirvn** [comment]
@uclaros , no you're not :) but that's a broader task to undertake. In the meantime, cleaning up is always good.

**Comment by nirvn** [comment]
(unrelated travis failure, task manager acting up)

**Pull request merged**